ocp_resources/data_import_cron.py

```diff
@@ -1,97 +1,85 @@
-            cert_configmap (str, optional): CertConfigMap provides a reference to the Registry certs
-            image_stream (str, optional): ImageStream is the name of image stream for import
-            bind_immediate_annotation (bool, optional): when WaitForFirstConsumer is set in StorageClass and the
-                DataSource should be bound immediately.
-            pull_method (str): can be either "pod" or "node" (node docker cache based import)
+            garbage_collect (str): GarbageCollect specifies whether old PVCs should be cleaned up after a
+              new PVC is imported. Options are currently "Outdated" and "Never",
+              defaults to "Outdated".
+
+            imports_to_keep (int): Number of import PVCs to keep when garbage collecting. Default is 3.
+
+            managed_data_source (str): ManagedDataSource specifies the name of the corresponding DataSource
+              this cron will manage. DataSource has to be in the same namespace.
+
+            retention_policy (str): RetentionPolicy specifies whether the created DataVolumes and
+              DataSources are retained when their DataImportCron is deleted.
+              Default is RatainAll.
+
+            schedule (str): Schedule specifies in cron format when and how often to look for new
+              imports
+
+            template (dict[str, Any]): Template specifies template for the DVs to be created
+
         """
         super().__init__(**kwargs)
-        self.image_stream = image_stream
-        self.url = url
-        self.cert_configmap = cert_configmap
-        self.pull_method = pull_method
-        self.storage_class = storage_class
-        self.size = size
-        self.schedule = schedule
+
         self.garbage_collect = garbage_collect
-        self.managed_data_source = managed_data_source
         self.imports_to_keep = imports_to_keep
-        self.bind_immediate_annotation = bind_immediate_annotation
+        self.managed_data_source = managed_data_source
+        self.retention_policy = retention_policy
+        self.schedule = schedule
+        self.template = template
 
     def to_dict(self) -> None:
         super().to_dict()
+
         if not self.kind_dict and not self.yaml_file:
-            if self.image_stream and self.url:
-                raise ValueError("imageStream and url cannot coexist")
-
-            if not self.pull_method:
-                raise MissingRequiredArgumentError(argument="pull_method")
-
-            self.res.update({
-                "spec": {
-                    "template": {"spec": {"source": {"registry": {"pullMethod": self.pull_method}}}},
-                }
-            })
-            spec = self.res["spec"]["template"]["spec"]
-
-            if self.bind_immediate_annotation:
-                self.res["metadata"].setdefault("annotations", {}).update({
-                    f"{NamespacedResource.ApiGroup.CDI_KUBEVIRT_IO}/storage.bind.immediate.requested": ("true")
-                })
```

Comment:
We need an option to add this annotation... 
@myakove, can we have it in the wrapper? What is the right way to do it?